codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpBoundProtocolGenerator.kt

```diff
@@ -237,18 +237,16 @@ class ServerHttpBoundProtocolTraitImplGenerator(
         }
         val verifyAcceptHeaderStaticContentTypeInit = writable {
             httpBindingResolver.responseContentType(operationShape)?.also { contentType ->
-                val init = when (contentType) {
-                    "application/json" -> "const $staticContentType: #{Mime}::Mime = #{Mime}::APPLICATION_JSON;"
-                    "application/octet-stream" -> "const $staticContentType: #{Mime}::Mime = #{Mime}::APPLICATION_OCTET_STREAM;"
-                    "application/x-www-form-urlencoded" -> "const $staticContentType: #{Mime}::Mime = #{Mime}::APPLICATION_WWW_FORM_URLENCODED;"
-                    else ->
-                        """
-                        static $staticContentType: #{OnceCell}::sync::Lazy<#{Mime}::Mime> = #{OnceCell}::sync::Lazy::new(|| {
-                            ${contentType.dq()}.parse::<#{Mime}::Mime>().expect("BUG: MIME parsing failed, content_type is not valid")
-                        });
-                        """
-                }
-                rustTemplate(init, *codegenScope)
+                val (type, subtype) = contentType.split('/')
+                rustTemplate(
+                    """
+                    const $staticContentType: #{MediaType}::MediaType = #{MediaType}::MediaType::new(
```

Comment:
Why isn't this a change in behavior?

Reply:
I think this is the same behavior, just simplified logic. Here's some code that is generated for the `pokemon-serivce-server-sdk` example.


`CONTENT_TYPE_GETSTORAGE` Before:
```
const CONTENT_TYPE_GETSTORAGE: ::mime::Mime = ::mime::APPLICATION_JSON;
```

`CONTENT_TYPE_GETSTORAGE` After:
```
const CONTENT_TYPE_GETSTORAGE: ::mediatype::MediaType = ::mediatype::MediaType::new(
    ::mediatype::Name::new_unchecked("application"),
    ::mediatype::Name::new_unchecked("json"),
);
```

`CONTENT_TYPE_CAPTUREPOKEMON` Before:
```
static CONTENT_TYPE_CAPTUREPOKEMON: ::once_cell::sync::Lazy<::mime::Mime> =
    ::once_cell::sync::Lazy::new(|| {
        "application/vnd.amazon.eventstream"
            .parse::<::mime::Mime>()
            .expect("BUG: MIME parsing failed, content_type is not valid")
    });
```
`CONTENT_TYPE_CAPTUREPOKEMON` After:
```
const CONTENT_TYPE_CAPTUREPOKEMON: ::mediatype::MediaType = ::mediatype::MediaType::new(
    ::mediatype::Name::new_unchecked("application"),
    ::mediatype::Name::new_unchecked("vnd.amazon.eventstream"),
);
```


For the `mime` crate, only certain predefined constants were const. For an arbitrary content type string, we had to use a static once_cell. For the `mediatype` crate all content types are const constructible, so we don't have to use  once_cell anymore or handle special cases differently.

Reply:
Your point is fine in the case in which the content-type is valid. Right now the string you can see is exactly one of:
```
"application/json"
"application/octet-stream"
"application/vnd.amazon.eventstream"
"application/x-amz-json-1.0"
"application/x-amz-json-1.1"
"application/xml"
"image/jpeg"
"text/plain"
```
What happens if you find an invalid content-type, like `type//x`? 
Currently your implementation is:
```
::mediatype::MediaType::new(
        ::mediatype::Name::new_unchecked("type"),
        ::mediatype::Name::new_unchecked(""),
    );
```
which constructs an invalid `MediaType`: `MediaType { ty: Name("type"), subty: Name(""), suffix: None, params: [] }`.

Your implementation, not to change behavior, should be:
```
::mediatype::MediaType::new(
        ::mediatype::Name::new("...").expect("..."),
        ::mediatype::Name::new("...").expect("..."),
    );
```
which parses the string and panics, as did the previous implementation when `mime::parse` failed.
> all content types are const constructible, so we don't have to use once_cell anymore or handle special cases differently

If this is the case also for invalid strings, it's not worth it - or it has to be done in Kotlin.

Reply:
Yea `mediatype::Name::new()` is not const while `mediatype::Name::new_unchecked()` is const. So there is no way to handle invalid strings using the const constructions from the `mediatype` crate. I think it's best to go back to the way it was previously written. That is, handle certain common cases with const constructors and handle the general case with a once_cell. 

```
when (contentType) {
                    "application/json" -> "const $staticContentType: #{MediaType}::MediaType = #{MediaType}::media_type!(APPLICATION/JSON);"
                    "application/octet-stream" -> "const $staticContentType: #{MediaType}::MediaType = #{MediaType}::media_type!(APPLICATION/OCTET_STREAM);"
                    "application/x-www-form-urlencoded" -> 
                        """
                        const $staticContentType: #{MediaType}::MediaType = #{MediaType}::mediatype::new(
                            #{MediaType}::names::APPLICATION, 
                            #{MediaType}::Name::new_unchecked(\"x-www-form-urlencoded\")
                        );
                        """
                    else ->
                        """
                        static $staticContentType: #{OnceCell}::sync::Lazy<#{MediaType}::MediaType> = #{OnceCell}::sync::Lazy::new(|| {
                            #{MediaType}::MediaType::parse(${contentType.dq()}).expect("BUG: MIME parsing failed, content_type is not valid")
                        });
                        """
                }
```